From: Hans de Goede <johannes.goede@oss.qualcomm.com>
Date: Thu, 3 Sep 2026 22:52:37 +0200
Subject: fix(drm): add leds-qcom-lpg to aarch64 specific modules needed by
 drm

leds-qcom-lpg is not only a LED class driver but it can also be a pwm class
driver and sometimes its pwm function is used for panel backlight control
in which case it is needed for drm/kms to work.

Since leds-qcom-lpg lives under drivers/leds/rgb/ it is not automatically
picked-up by the existing 'instmods "=drivers/pwm"', add it explicitly
to the instmods argument to get it included into the initramfs.

This follows the pattern of how the amdkfd and hyperv_fb special cases are
already handled.

Signed-off-by: Hans de Goede <johannes.goede@oss.qualcomm.com>
Origin: upstream, https://github.com/dracut-ng/dracut/commit/6fe62db59b8a450319a36f062d396f9bccfd9bb5
---
 modules.d/45drm/module-setup.sh | 1 +
 1 file changed, 1 insertion(+)

diff --git a/modules.d/45drm/module-setup.sh b/modules.d/45drm/module-setup.sh
index 17f7ebc..9ee64ba 100755
--- a/modules.d/45drm/module-setup.sh
+++ b/modules.d/45drm/module-setup.sh
@@ -12,6 +12,7 @@ installkernel() {
     if [[ ${DRACUT_ARCH} == arm* || ${DRACUT_ARCH} == aarch64 ]]; then
         # arm/aarch64 specific modules needed by drm
         hostonly=$(optional_hostonly) instmods \
+            leds-qcom-lpg \
             "=drivers/gpu/drm/i2c" \
             "=drivers/gpu/drm/panel" \
             "=drivers/gpu/drm/bridge" \
